diff --git a/include/linux/sched.h b/include/linux/sched.h
--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -1523,7 +1523,6 @@
 	 */
 	struct callback_head		l1d_flush_kill;
 #endif
-
 #ifdef CONFIG_SLIM_SCHED
 	ANDROID_KABI_USE(1, unsigned long sched_prop);
 	ANDROID_KABI_USE(2, struct sched_ext_entity *scx);
@@ -1934,6 +1933,7 @@
 	return PRIO_TO_NICE((p)->static_prio);
 }
 
+
 extern int can_nice(const struct task_struct *p, const int nice);
 extern int task_curr(const struct task_struct *p);
 extern int idle_cpu(int cpu);
diff --git a/include/linux/sched/ext.h b/include/linux/sched/ext.h
--- a/include/linux/sched/ext.h
+++ b/include/linux/sched/ext.h
@@ -569,6 +569,8 @@
 	u16				demand_scaled;
 	void 			*sdsq;
 };
+
+
 
 /*
  * The following is embedded in task_struct and contains all fields necessary
diff --git a/kernel/sched/core.c b/kernel/sched/core.c
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -187,6 +187,7 @@
 	if (p->sched_class == &ext_sched_class)
 		return MAX_RT_PRIO + MAX_NICE + 1; /* 120, squash ext */
 #endif
+
 	return MAX_RT_PRIO + MAX_NICE; /* 119, squash fair */
 }
 
@@ -3771,7 +3772,6 @@
 	 * [ this allows ->select_task() to simply return task_cpu(p) and
 	 *   not worry about this generic constraint ]
 	 */
-
 	if (unlikely(!is_cpu_allowed(p, cpu)))
 		cpu = select_fallback_rq(task_cpu(p), p);
 
@@ -4891,6 +4891,7 @@
  */
 int sched_fork(unsigned long clone_flags, struct task_struct *p)
 {
+
 	int ret;
 
 	trace_android_rvh_sched_fork(p);
@@ -5009,6 +5010,7 @@
 void sched_post_fork(struct task_struct *p)
 {
 	uclamp_post_fork(p);
+
 	scx_post_fork(p);
 }
 
@@ -5887,6 +5889,7 @@
 	}
 #endif
 	trace_android_vh_scheduler_tick(rq);
+
 }
 
 #ifdef CONFIG_NO_HZ_FULL
@@ -6913,6 +6916,7 @@
 		psi_sched_switch(prev, next, !task_on_rq_queued(prev));
 
 		trace_sched_switch(sched_mode & SM_MASK_PREEMPT, prev, next, prev_state);
+
 		/* Also unlocks the rq: */
 		rq = context_switch(rq, prev, next, &rf);
 	} else {
@@ -7383,11 +7387,10 @@
 		} else {
 			if (dl_prio(oldprio))
 				p->dl.pi_se = &p->dl;
-			if (rt_prio(oldprio))
+			else if (rt_prio(oldprio))
 				p->rt.timeout = 0;
 			else if (!task_has_idle_policy(p))
 				reweight_task(p, prio - MAX_RT_PRIO);
-
 		}
 
 		__setscheduler_prio(p, prio);
@@ -7874,6 +7877,7 @@
 	struct rq *rq;
 	bool cpuset_locked = false;
 
+
 	/* The pi code expects interrupts enabled */
 	BUG_ON(pi && in_interrupt());
 recheck:
@@ -7938,6 +7942,7 @@
 	 * To be able to change p->policy safely, the appropriate
 	 * runqueue lock must be held.
 	 */
+
 	rq = task_rq_lock(p, &rf);
 	update_rq_clock(rq);
 
@@ -8009,6 +8014,7 @@
 	if (unlikely(oldpolicy != -1 && oldpolicy != p->policy)) {
 		policy = oldpolicy = -1;
 		task_rq_unlock(rq, p, &rf);
+
 		if (cpuset_locked)
 			cpuset_unlock();
 		goto recheck;
@@ -8066,6 +8072,7 @@
 	preempt_disable();
 	head = splice_balance_callbacks(rq);
 	task_rq_unlock(rq, p, &rf);
+
 	if (pi) {
 		if (cpuset_locked)
 			cpuset_unlock();
@@ -8080,6 +8087,7 @@
 
 unlock:
 	task_rq_unlock(rq, p, &rf);
+
 	if (cpuset_locked)
 		cpuset_unlock();
 	return retval;
@@ -10214,6 +10222,7 @@
 	BUG_ON(!sched_class_above(&fair_sched_class, &ext_sched_class));
 	BUG_ON(!sched_class_above(&ext_sched_class, &idle_sched_class));
 #endif
+
 	wait_bit_init();
 
 #ifdef CONFIG_FAIR_GROUP_SCHED
@@ -10854,7 +10863,6 @@
 	struct task_group *tg = css_tg(css);
 	struct task_group *parent = css_tg(css->parent);
 
-
 	if (parent)
 		sched_online_group(tg, parent);
 
@@ -10910,6 +10918,7 @@
 	struct cgroup_subsys_state *css;
 
 	cgroup_taskset_for_each(task, css, tset) {
+
 		sched_move_task(task);
 	}
 
@@ -11591,6 +11600,7 @@
 		.write_u64 = cpu_uclamp_ls_write_u64,
 	},
 #endif
+
 	{ }	/* Terminate */
 };
 
diff --git a/kernel/sched/ext.c b/kernel/sched/ext.c
--- a/kernel/sched/ext.c
+++ b/kernel/sched/ext.c
@@ -284,6 +284,8 @@
 	__ret;									\
 })
 
+//#include "./slim_walt.c"
+
 /* @mask is constant, always inline to cull unnecessary branches */
 static __always_inline bool scx_kf_allowed(u32 mask)
 {
@@ -602,6 +604,7 @@
 		container_of(node_a, struct sched_ext_entity, dsq_node.priq);
 	const struct sched_ext_entity *b =
 		container_of(node_b, struct sched_ext_entity, dsq_node.priq);
+
 	return time_before64(a->dsq_vtime, b->dsq_vtime);
 }
 
@@ -674,6 +677,7 @@
 	} else {
 		list_del_init(&p->scx->dsq_node.fifo);
 	}
+
 }
 
 static bool task_linked_on_dsq(struct task_struct *p)
@@ -1015,6 +1019,9 @@
 		SCX_CALL_OP_TASK(SCX_KF_REST, stopping, p, false);
 	}
 
+	//if(task_current(rq, p))
+	//	scx_update_task_ravg(p, rq, PUT_PREV_TASK, rq->clock);
+
 	if (SCX_HAS_OP(quiescent))
 		SCX_CALL_OP_TASK(SCX_KF_REST, quiescent, p, deq_flags);
 
@@ -1648,6 +1655,9 @@
 	/* see dequeue_task_scx() on why we skip when !QUEUED */
 	if (SCX_HAS_OP(running) && (p->scx->flags & SCX_TASK_QUEUED))
 		SCX_CALL_OP_TASK(SCX_KF_REST, running, p);
+
+	//if(p->scx->flags & SCX_TASK_QUEUED)
+	//	scx_update_task_ravg(p, rq, PICK_NEXT_TASK, rq->clock);
 
 	watchdog_unwatch_task(p, true);
 
@@ -1705,6 +1715,10 @@
 	if (SCX_HAS_OP(stopping) && (p->scx->flags & SCX_TASK_QUEUED))
 		SCX_CALL_OP_TASK(SCX_KF_REST, stopping, p, true);
 
+	//if(p->scx->flags & SCX_TASK_QUEUED)
+	//	scx_update_task_ravg(p, rq, PUT_PREV_TASK, rq->clock);
+
+
 	/*
 	 * If we're being called from put_prev_task_balance(), balance_scx() may
 	 * have decided that @p should keep running.
@@ -2140,7 +2154,7 @@
 static void task_tick_scx(struct rq *rq, struct task_struct *curr, int queued)
 {
 	update_curr_scx(rq);
-
+	//scx_update_task_ravg(curr, rq, TASK_UPDATE, rq->clock);
 	/*
 	 * While disabling, always resched and refresh core-sched timestamp as
 	 * we can't trust the slice management or ops.core_sched_before().
@@ -2175,6 +2189,7 @@
 			return ret;
 		}
 	}
+	//scx_sched_init_task(p);
 
 	if (p->scx->disallow) {
 		struct rq *rq;
@@ -2796,6 +2811,8 @@
 		ret = -EBUSY;
 		goto err_unlock;
 	}
+
+	//slim_walt_enable(true);
 
 	/*
 	 * Set scx_ops, transition to PREPPING and clear exit info to arm the
diff --git a/kernel/sched/ext.h b/kernel/sched/ext.h
--- a/kernel/sched/ext.h
+++ b/kernel/sched/ext.h
@@ -153,12 +153,14 @@
 	__scx_notify_pick_next_task(rq, p, active);
 }
 
+//extern void scx_scheduler_tick(void);
 static inline void scx_notify_sched_tick(void)
 {
 	unsigned long last_check;
 
 	if (!scx_enabled())
 		return;
+	//scx_scheduler_tick();
 
 	last_check = scx_watchdog_timestamp;
 	if (unlikely(time_after(jiffies, last_check + scx_watchdog_timeout))) {
diff --git a/kernel/time/tick-sched.c b/kernel/time/tick-sched.c
--- a/kernel/time/tick-sched.c
+++ b/kernel/time/tick-sched.c
@@ -1124,7 +1124,6 @@
 	ktime_t expires;
 
 	trace_android_vh_tick_nohz_idle_stop_tick(NULL);
-
 	/*
 	 * If tick_nohz_get_sleep_length() ran tick_nohz_next_event(), the
 	 * tick timer expiration time is known already.
