diff --git a/target_if/mlme/vdev_mgr/src/target_if_vdev_mgr_rx_ops.c b/target_if/mlme/vdev_mgr/src/target_if_vdev_mgr_rx_ops.c
index 7c4a4191..bb98a4ec 100644
--- a/target_if/mlme/vdev_mgr/src/target_if_vdev_mgr_rx_ops.c
+++ b/target_if/mlme/vdev_mgr/src/target_if_vdev_mgr_rx_ops.c
@@ -41,6 +41,7 @@
 #include <cdp_txrx_ctrl.h>
 #endif
 #include <target_if_psoc_timer_tx_ops.h>
+#include "wma_frame_inject.h"
 
 static inline
 void target_if_vdev_mgr_handle_recovery(struct wlan_objmgr_psoc *psoc,
@@ -424,6 +425,10 @@ static int target_if_vdev_mgr_start_response_handler(ol_scn_t scn,
 	}
 
 	vdev_id = vdev_start_resp.vdev_id;
+	if (wma_mon_inject_vdev_start_response(vdev_id,
+					       vdev_start_resp.status))
+		return 0;
+
 	vdev_rsp = rx_ops->psoc_get_vdev_response_timer_info(psoc, vdev_id);
 	if (!vdev_rsp) {
 		mlme_err("vdev response timer is null VDEV_%d PSOC_%d",
@@ -513,6 +518,9 @@ static int target_if_vdev_mgr_stop_response_handler(ol_scn_t scn,
 		return -EINVAL;
 	}
 
+	if (wma_mon_inject_vdev_stop_response(vdev_id))
+		return 0;
+
 	vdev_rsp = rx_ops->psoc_get_vdev_response_timer_info(psoc, vdev_id);
 	if (!vdev_rsp) {
 		mlme_err("vdev response timer is null VDEV_%d PSOC_%d",
@@ -575,6 +583,9 @@ static int target_if_vdev_mgr_delete_response_handler(ol_scn_t scn,
 		return -EINVAL;
 	}
 
+	if (wma_mon_inject_vdev_delete_response(vdev_del_resp.vdev_id))
+		return 0;
+
 	vdev_rsp = rx_ops->psoc_get_vdev_response_timer_info(psoc,
 							 vdev_del_resp.vdev_id);
 	if (!vdev_rsp) {
